src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogit.scala

```diff
@@ -56,9 +56,13 @@ class ModeChoiceMultinomialLogit(
     } else {
       val modeCostTimeTransfers = altsToModeCostTimeTransfers(alternatives, attributesOfIndividual, destinationActivity)
 
-      val bestInGroup = modeCostTimeTransfers groupBy (_.embodiedBeamTrip.tripClassifier) map {
-        case (_, group) => findBestIn(group)
-      }
+      val bestInGroup: immutable.Iterable[ModeCostTimeTransfer] =
+        modeCostTimeTransfers
+          .groupBy(_.embodiedBeamTrip.tripClassifier)
+          .flatMap {
+            case (_, group: IndexedSeq[ModeCostTimeTransfer]) if group.isEmpty => None
```

Comment:
as you can see it is a huge change. maybe better just create a new PR for those refactoring. otherwise does not finish. there is another PR just to remove 90 messages of scapegoat